src/natcap/invest/habitat_quality.py

```diff
@@ -895,13 +962,35 @@ def _compute_rarity_operation(
             code_index[code] = 0.0
 
     pygeoprocessing.reclassify_raster(
-        new_cover_path, code_index, rarity_path, gdal.GDT_Float32,
+        new_cover_path, code_index, rarity_raster_path, gdal.GDT_Float32,
         _OUT_NODATA)
 
+    _generate_rarity_csv(code_index, rarity_csv_path)
+
     LOGGER.info('Finished rarity computation on'
                 f' {os.path.basename(lulc_path_band[0])} land cover.')
 
 
+def _generate_rarity_csv(rarity_dict, target_csv_path):
+    """Generate CSV containing rarity values by LULC code.
+
+    Args:
+        rarity_dict (dict): dictionary containing LULC codes (as keys)
+            and their associated rarity values (as values).
+        target_csv_path (string): path to output CSV.
+
+    Returns:
+        None
+    """
+    lulc_codes = sorted(rarity_dict)
+    cols = ['LULC_code', 'rarity_value']
```

Comment:
As mentioned above, I believe we've started to standardize lowercase for table column headers. Even though LULC is an acronym, I think it should still follow that standard.